pkg/graveler/graveler.go

```diff
@@ -1489,15 +1499,19 @@ func (g *Graveler) Commit(ctx context.Context, repositoryID RepositoryID, branch
 			parentGeneration = commit.Generation
 		}
 		commit.Generation = parentGeneration + 1
-		changes, err := g.StagingManager.List(ctx, branch.StagingToken, ListingMaxBatchSize)
-		if err != nil {
-			return "", fmt.Errorf("staging list: %w", err)
-		}
-		defer changes.Close()
+		if params.SourceMetaRange != nil {
```

Comment:
Not part of the changes but suggest to change the name of the variable 'commit' (line 1494) as it shadows an already defined variable in outer scope.

Reply:
Done.